shared/middleware/net_sim_api.rb

```diff
@@ -78,6 +119,21 @@ def has_json_utf8_headers(request)
     dont_cache
     table = get_table(shard_id, table_name)
     int_id = id.to_i
+
+    if table_name == TABLE_NAMES[:node]
+      # Cascade deletions
+      wire_ids = delete_wires_for_node(shard_id, int_id)
+      message_ids = delete_messages_for_node(shard_id, int_id)
+
+      unless wire_ids.empty?
+        get_pub_sub_api.publish(shard_id, TABLE_NAMES[:wire], {:action => 'delete_many', :ids => wire_ids})
```

Comment:
Is it really necessary to publish multiple invalidations? This will result in additional traffic and perhaps unnecessary refreshes on the client. 

It seems like you could combine all of the information into a single message (e.g. with :wire_ids and :message_ids fields), or just don't send those fields to the client if the client doesn't care about them.  (which would keep message size down)